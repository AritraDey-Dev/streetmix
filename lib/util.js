exports.parseLoginToken = function (req) {
  var authHeader = req.header('Authorization')

  if (!authHeader) {
    return null
  }

  var auth = authHeader.split(/\s+/)

  if (auth.length < 3) {
    return null
  }

  var authScheme = auth.shift()
  var realm, loginToken

  for (var authParamIndex in auth) {
    var authParam = auth[authParamIndex]
    var kv = authParam.split('=')
    var authParamKey = kv[0]
    var authParamValue = kv[1]

    switch (authParamKey) {
      case 'realm':
        realm = authParamValue.replace(/(^"|"$)/g, '')
        break
      case 'loginToken':
        loginToken = authParamValue.replace(/(^"|"$)/g, '')
        break
    }
  }

  if ((authScheme !== 'Streetmix') || (realm !== '')) {
    return null
  }

  return loginToken
}

exports.ERRORS = {
  USER_NOT_FOUND: 'USER_NOT_FOUND',
  INTERNAL_ERROR: 'INTERNAL_ERROR',
  STREET_NOT_FOUND: 'STREET_NOT_FOUND',
  STREET_DELETED: 'STREET_DELETED',
  UNAUTHORISED_ACCESS: 'UNAUTHORISED_ACCESS',
  FORBIDDEN_REQUEST: 'FORBIDDEN_REQUEST',
  CANNOT_CREATE_STREET: 'CANNOT_CREATE_STREET'
}
