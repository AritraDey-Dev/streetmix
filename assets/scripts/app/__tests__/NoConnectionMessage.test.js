/* eslint-env jest */
import React from 'react'
import { fireEvent } from '@testing-library/react'
import { renderWithReduxAndIntl } from '../../../../test/helpers/render'
import NoConnectionMessage from '../NoConnectionMessage'
import { nonblockingAjaxTryAgain } from '../../util/fetch_nonblocking'
import { showNoConnectionMessage } from '../../store/actions/status'

jest.mock('../../util/fetch_nonblocking', () => ({
  nonblockingAjaxTryAgain: jest.fn()
}))

const initialState = {
  status: {
    noConnectionMessage: false
  }
}

describe('NoConnectionMessage', () => {
  it('tries to reconnect when button is clicked', () => {
    const wrapper = renderWithReduxAndIntl(<NoConnectionMessage />, {
      initialState
    })
    fireEvent.click(wrapper.getByText('Try again'))
    expect(nonblockingAjaxTryAgain).toHaveBeenCalled()
  })

  it('does not have the visibility class when mounted', () => {
    const wrapper = renderWithReduxAndIntl(<NoConnectionMessage />, {
      initialState
    })
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('applies visibility class when scheduled', () => {
    // Make setTimeouts run automatically
    jest.useFakeTimers()

    const wrapper = renderWithReduxAndIntl(<NoConnectionMessage />, {
      initialState
    })

    // Schedule the message to appear
    wrapper.store.dispatch(showNoConnectionMessage(true))

    // Run timers, then checking visibility
    jest.runAllTimers()
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it.todo('removes visibility class when connectivity returns')
})
