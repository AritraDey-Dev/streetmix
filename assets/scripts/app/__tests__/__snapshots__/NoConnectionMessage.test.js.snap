// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoConnectionMessage applies visibility class when scheduled 1`] = `
<DocumentFragment>
  <div
    class="status-message status-message-visible"
  >
    <div
      class="status-message-content"
    >
      Streetmix is having trouble connecting to the Internet.
      <button
        tabindex="0"
      >
        Try again
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NoConnectionMessage does not have the visibility class when mounted 1`] = `
<DocumentFragment>
  <div
    class="status-message"
  >
    <div
      class="status-message-content"
    >
      Streetmix is having trouble connecting to the Internet.
      <button
        tabindex="-1"
      >
        Try again
      </button>
    </div>
  </div>
</DocumentFragment>
`;
